test/Garnet.test/RespListGarnetClientTests.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Garnet.client;
+using NUnit.Framework;
+
+namespace Garnet.test
+{
+    [TestFixture]
+    public class RespListGarnetClientTests
+    {
+        private GarnetServer _server;
+
+        [OneTimeSetUp]
+        public void Setup()
+        {
+            TestUtils.DeleteDirectory(TestUtils.MethodTestDir, wait: true);
+            _server = TestUtils.CreateGarnetServer(TestUtils.MethodTestDir, lowMemory: true);
+            _server.Start();
+        }
+
+        [Test]
+        public void AddElementsToTheListHead_WithCallback()
```

Comment:
Can use test cases here too, can save some code duplication and make the test more readable

Reply:
I tried this

```csharp
[Test]
[TestCaseSource(nameof(_list1))]
public void AddElementsToTheListHead_WithCallback(string key, List<string> elements, int result)
{
    // Arrange
    using var db = new GarnetClient(TestUtils.Address, TestUtils.Port);
    db.Connect();

    ManualResetEventSlim e = new();

    // Act & Assert
    db.ListLeftPush(key, elements, (_, returnValue, message) =>
    {
        Assert.AreEqual(result, returnValue);
        e.Set();
    });

    e.Wait();
    e.Reset();
    e.Dispose();
}
```

but the execution is hanging

Reply:
This works fine for me: 

```
public static object[] TestCases =
{
    new object[] { "list1", new List<string> { "foo" }, 1 },
    new object[] { "list1", new List<string> { "bar", "baz" }, 3 },
    new object[] { "list2", new List<string> { "foo", "baz" }, 2 },
};

[Test]
[TestCaseSource(nameof(TestCases))]
public void AddElementsToTheListHead_WithCallback(string key, List<string> elements, int result)
{
    // Arrange
    using var db = new GarnetClient(TestUtils.Address, TestUtils.Port);
    db.Connect();

    using ManualResetEventSlim e = new();

    // Act & Assert
    db.ListLeftPush(key, elements, (_, returnValue, _) =>
    {
        try
        {
            Assert.AreEqual(58, returnValue);
        }
        finally
        {
            e.Set();
        }
    });

    e.Wait();
}
```

Reply:
Thanks, I will try it, or next time you can commit directly to my PR

Reply:
This works fine for me:

```csharp
        public static object[] TestCases =
        {
            new object[] { "list1", new List<string> { "foo" }, 1L },
            new object[] { "list1", new List<string> { "bar", "baz" }, 3L },
            new object[] { "list2", new List<string> { "foo", "baz" }, 2L },
        };

        [Test]
        [TestCaseSource(nameof(TestCases))]
        public void AddElementsToTheListHead_WithCallback(string key, List<string> elements, long expectedResult)
        {
            // Arrange
            using var db = new GarnetClient(TestUtils.Address, TestUtils.Port);
            db.Connect();

            using ManualResetEventSlim e = new();

            var isResultSet = false;
            var actualResult = 0L;

            // Act & Assert
            db.ListLeftPush(key, elements, (_, returnValue, _) =>
            {
                actualResult = returnValue;
                isResultSet = true;
                e.Set();
            });

            e.Wait();
            Assert.IsTrue(isResultSet);
            Assert.AreEqual(expectedResult, actualResult);
        }
```

Reply:
FYI one of the 3 test is fail :(

Reply:
Why does it fail?

Reply:
I'm not sure